.github/workflows/test.yml

```
@@ -2,33 +1,47 @@
+
+    - name: Run tests
+      run: pytest --cov=./
 
-    - name: Report coverage to Codecov
+    - name: Upload Coverage to Codecov
       uses: codecov/codecov-action@v4
       with:
-        token: ${{ secrets.CODECOV_TOKEN }}
-        file: ./coverage.xml
-        fail_ci_if_error: true
+        token: ${{ secrets.CODECOV_TOKEN }}
```

yamllint

[error] no new line character at the end of file

(new-line-at-end-of-file)